### PR TITLE
Add Fedora VM support

### PR DESCRIPTION
**Pull request by monstermunchkin**

### CODE REVIEW

File: images/fedora.yaml [hunk 1]
```diff
@@ -36,6 +36,42 @@ targets:
...
+      - vm
+
+  - generator: fstab
+    types:
+      - vm
+
+  - path: /etc/machine-id
+    generator: dump
+    types:
+      - vm
```
Comment (stgraber) [inline images/fedora.yaml]: Any reason not to do it for containers too, same as we do for Debian/Ubuntu?

File: images/fedora.yaml [hunk 2]
```diff
@@ -140,6 +180,16 @@ packages:
       - sssd-client
      action: remove
 
+   - packages:
+     - grub2-efi-x64
+     - kernel
+     - shim
+     - btrfs-progs
```
Comment (stgraber) [inline images/fedora.yaml]: Hmm, why btrfs?

File: images/fedora.yaml (hunk 1 above)
Comment (stgraber) [reply images/fedora.yaml]: Actually, given you're now always generating a machine-id as part of the image, we absolutely need to clear the file in both containers and vms, or we'll end up with all our containers having the same unique id.

File: images/fedora.yaml (hunk 2 above)
Comment (monstermunchkin) [reply images/fedora.yaml]: IIRC something complained about btrfs missing.

File: images/fedora.yaml (hunk 1 above)
Comment (monstermunchkin) [reply images/fedora.yaml]: Yeah, that's wrong. I'll fix that.

File: images/fedora.yaml (hunk 2 above)
Comment (monstermunchkin) [reply images/fedora.yaml]: I'll test this again.

Comment (monstermunchkin) [reply images/fedora.yaml]: I removed it. I had a reason for adding it but I don't remember why. Anyway, it works fine without it.

**Pull request merged**